hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/keygen/CustomKeyGenerator.java

```diff
@@ -157,12 +155,6 @@ private String getPartitionPath(Option<GenericRecord> record, Option<Row> row, O
     return partitionPath.toString();
   }
 
-  private void validateRecordKeyFields() {
-    if (getRecordKeyFieldNames() == null || getRecordKeyFieldNames().isEmpty()) {
-      throw new HoodieKeyException("Unable to find field names for record key in cfg");
-    }
```

Comment:
If we know the user set up `UPSERT` operation explicitly, we can keep this validation. Or maybe we just move the validation to the kengen factory class.